packages/models/src/resource-binding/manager.ts

```diff
@@ -0,0 +1,201 @@
+  #lastBlock: Block | undefined = undefined
+  #tipBlockNumber = BI.from(0)
+
+  constructor(private _listener: Listener<Header>, private _dataSource: ChainSource) {
+    super()
+  }
+
+  onListenBlock = (blockHeader: Header) => {
+    const currentBlockNumber = BI.from(blockHeader.number)
+    if (currentBlockNumber.gt(this.tipBlockNumber)) {
+      this.tipBlockNumber = currentBlockNumber
+    }
+  }
+
+  private update(pollingInterval = 1000) {
+    return setInterval(async () => {
+      if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+        this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+      }
+    }, pollingInterval)
+  }
+
+  private updateStore(block: Block) {
+    if (!this.#lastBlock || BI.from(block.header.number).gt(BI.from(this.#lastBlock.header.number))) {
+      const changes = this.filterCells(block)
+      for (const [store, input, data] of changes) {
+        const upgrade: { witness: HexString; cell: Cell }[] = []
+        for (const [cell, witness] of data) {
+          upgrade.push({ witness, cell })
+          const outPoint = cell.outPoint
+          if (outPoint) {
+            this.#registryOutpoint.set(outPointToOutPointString(outPoint), store)
+          }
+        }
+        this.sendMessage(store, 'update_cells', upgrade)
+
+        this.sendMessage(
+          store,
+          'remove_cell',
+          input.map((v) => v.previousOutput),
+        )
+      }
+      this.#lastBlock = block
+    }
+  }
+
+  private filterOutputs(tx: Transaction, block: Block): Map<OutPointString, [Cell, string]> {
+    const outputs = new Map<OutPointString, [Cell, string]>()
+    for (const outputIndex in tx.outputs) {
+      if (tx.hash) {
+        const outPoint = {
+          txHash: tx.hash,
+          index: BI.from(outputIndex).toHexString(),
+        }
+        outputs.set(outPointToOutPointString(outPoint), [
+          {
+            cellOutput: tx.outputs[outputIndex],
+            data: tx.outputsData[outputIndex],
+            outPoint,
+            blockHash: block.header.hash,
+            blockNumber: block.header.number,
+          },
+          tx.witnesses[outputIndex],
+        ])
+      }
+    }
+
+    return outputs
+  }
+
+  private filterCells(block: Block): [ResourceBindingRegistry, Input[], [Cell, string][]][] {
+    const changes: Map<ActorURI, [ResourceBindingRegistry, Input[], [Cell, string][]]> = new Map()
+    let outputs = new Map<OutPointString, [Cell, string]>()
+    for (const tx of block.transactions) {
+      outputs = new Map([...outputs, ...this.filterOutputs(tx, block)])
+      for (const input of tx.inputs) {
+        const outPointString = outPointToOutPointString(input.previousOutput)
+        if (outputs.has(outPointString)) {
+          outputs.delete(outPointString)
+        } else {
+          const registry = this.#registryOutpoint.get(outPointString)
+          if (registry) {
```

Comment:
Please add the `else-branch` with `// ignore` etc. to imply that the `else` branch is neglectable by design

Reply:
This `else-branch` is necessary, because if the outpoint of the `input` is not in the filtered `output`s, it must be sent to the `Store` before.

Reply:
Seems that you're talking about the `else branch` of `if statement` at line 93 https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR93

The one I asked about is of `if statement` line 99 https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR99, it doesn't have an `else branch,` but I was not clear why it could be ignored. So I would like it to be annotated.

Reply:
> Seems that you're talking about the `else branch` of `if statement` at line 93 https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR93
> 
> The one I asked about is of `if statement` line 99 https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR99, it doesn't have an `else branch,` but I was not clear why it could be ignored. So I would like it to be annotated.

From my understanding, an input cell, if the current block doesn't output it, it must be output by a former block. If it isn't registered by any `Store`s, then it certainly belongs to any other applications.

Reply:
> > Seems that you're talking about the `else branch` of `if statement` at line 93 [#100 (files)](https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR93)
> > The one I asked about is of `if statement` line 99 [#100 (files)](https://github.com/ckb-js/kuai/pull/100/files#diff-6beced91eeabe7f90f6b66349a7a50e5a2e7cac2c73c536c6c32131c8082ddcdR99), it doesn't have an `else branch,` but I was not clear why it could be ignored. So I would like it to be annotated.
> 
> From my understanding, an input cell, if the current block doesn't output it, it must be output by a former block. If it isn't registered by any `Store`s, then it certainly belongs to any other applications.

Got it